src/structs/model.rs

```diff
@@ -364,7 +364,8 @@ impl ModelPerColor {
 
         let mut coef = 0;
         if length != 0 {
-            let sign = &mut self.sign_counts[calc_sign_index(best_prior)][best_prior_bit_len];
+            let sign =
+                &mut self.sign_counts[calc_sign_index(best_prior as i16)][best_prior_bit_len];
```

Comment:
best_prior should be passed in as an i16 from the beginning, since unsigned_abs will work differently as well

wait... ignore looks like context stores an int

Reply:
Yep, it's a mess.